content/technotes/2019-10-09-cran-checks-api-update.md

```diff
@@ -0,0 +1,112 @@
+
+It's great that packages are being tested in a variety of scenarios to make sure each package will work for as many users as possible.
+
+The only drawback to CRAN checks is that they are not available in a machine readable way. When information is made available as machine readable many additional things become possible.
+
+## CRAN Checks API
+
+To make CRAN check data available in a machine readable way I decided to build an API. Two years ago I [introduced the cran checks API](https://recology.info/2017/09/cranchecks-api/) on my blog.
+
+Over the two years the API has gotten some tweaks and fixes, some of which are very recent.
+
+When the API first came out the only routes were:
+
+- `/heartbeat`
+- `/docs`
+- `/pkgs`
+- `/pkgs/:pkg_name:`
+
+Now the routes are:
+
+- `/heartbeat`
+- `/docs`
+- `/pkgs`
+- `/pkgs/:pkg_name:`
+- `/pkgs/:pkg_name:/history`
+- `/history/:date`
+- `/maintainers`
+- `/maintainers/:email:`
+- `/badges/:type/:package`
+- `/badges/flavor/:flavor/:package`
+
+The big new additions since the beginning are maintainers, badges, and history
+
+### Maintainers
+
+The maintainers route gives you access to the data on the summary pages by package maintainer <https://cloud.r-project.org/web/checks/check_summary_by_maintainer.html>. `/maintainers` gives all maintainers data, while with `/maintainers/:email:` you can get data for a specific maintainer (by their email address).
+
+### Badges
+
+![](/img/blog-images/2019-10-09-cran-checks-api-update/svgs/ok.svg)
+
+![](/img/blog-images/2019-10-09-cran-checks-api-update/svgs/notok.svg)
+
+![](/img/blog-images/2019-10-09-cran-checks-api-update/svgs/note.svg)
+
+![](/img/blog-images/2019-10-09-cran-checks-api-update/svgs/warn.svg)
+
+![](/img/blog-images/2019-10-09-cran-checks-api-update/svgs/error.svg)
+
+![](/img/blog-images/2019-10-09-cran-checks-api-update/svgs/unknown.svg)
+
+Badges routes are by far the most widely used routes on this API, with ~900 R packages using a cran checks badge. Within the `/badges/:type/:package` route, you can get a badge for either `/badges/summary/:package` or `/badges/worst/:package`, where the former gives you a badge as `Not OK` (red) or `OK` (green), and the latter gives the worst check result (error, warn, note, and ok; each w/ a different color).
+
+The `/badges/flavor/:flavor/:package` route allows you to get a badge for a specific "flavor", that is, an operating system OR an R version. For example, `/badges/flavor/osx/ggplot2` or `/badges/flavor/release/ggplot2`.
+
+If you request a badges route with an uknown flavor or package you get an gray unknown badge (see above).
+
+### History
+
+We've had the `/pkgs/:pkg_name:/history` route for quite a while now, but it's been very slow to respond because the SQL database was not optimized. Its now fixed, and you can very quickly get up to the last 30 days of checks history. 
```

Comment:
```suggestion
We've had the `/pkgs/:pkg_name:/history` route for quite a while now, but it's been very slow to respond because the SQL database was not optimized. It's now fixed, and you can very quickly get up to the last 30 days of checks history. 
```

Its -> It's